docs/pages/rmf-audio-capture_L3_Test_Procedure.md

```diff
@@ -0,0 +1,170 @@
+    - [References](#references)
+  - [Level 3 Test Procedure](#level-3-test-procedure)
+    - [UT-Core functions to be added](#ut-core-functions-to-be-added)
+    - [Raft L3 Testing Requirements](#raft-l3-testing-requirements)
+    - [Input Stream Requirements](#input-stream-requirements)
+
+## Overview
+
+This document describes the level 3 testing suite for the RMF_AudioCapture module.
+
+### Acronyms, Terms and Abbreviations
+
+- `HAL` \- Hardware Abstraction Layer, may include some common components
+- `HAL.h`  \- Abstracted defined API to control the hardware
+- `HAL.c`  \- Implementation wrapper layer created by the `OEM` or `SoC` Vendor.
+- `RDK`  \- Reference Design Kit for All Devices
+- `RDK-V`  \- Reference Design Kit for Video Devices
+- `UT`  \- Unit Test(s)
+- `OEM`  \- Original Equipment Manufacture (Sky is also an OEM)
+- `SoC`  \- System on a Chip
+
+### Definitions
+
+- `Broadcom` \- `SoC` manufacturer <https://www.broadcom.com/>
+- `Amlogic` \- `SoC` manufacturer <https://en.wikipedia.org/wiki/Amlogic>
+- `Realtek` \- `SoC` manufacturer <https://en.wikipedia.org/wiki/Realtek>
+- `Soc Vendor` \- Definition to encompass multiple vendors
+- `Unit Tests` \- C Function tests that run on the target hardware
+- `Common Testing Framework` \- Off the shelf 3rd Party Testing Framework, or framework that does not require infrastructure to control it. That's not to say it cannot be controlled via infrastructure if required. Examples of which are.
+  - `GTest` \- Google Test Suit <https://google.github.io/googletest>
+  - `CUnit` \- C Testing Suit <http://cunit.sourceforge.net/>
+  - `Unity` \- C Embedded Testing Suit <http://www.throwtheswitch.org/unity>
+  - `ut-core` \- Common Testing Framework <https://github.com/comcast-sky/rdk-components-ut-core>, which wraps a open-source framework that can be expanded to the requirements for future comcast-sky framework.
+
+### References
+
+- `Feedback Loops` \- <https://www.softwaretestingnews.co.uk/4-methods-to-improve-your-feedback-loops-and-supercharge-your-testing-process/>
+- `Doxygen` \- SourceCode documentation tool - <https://www.doxygen.nl/index.html>
+- `Black Box Testing` \- <https://en.wikipedia.org/wiki/Black-box_testing>
+
+## Level 3 Test Procedure
+
+## UT-Core functions to be added
+
+- L3 - RMF_AudioCapture_Open - Depricated not being tested
+- L3 - RMF_AudioCapture_Open_Type(Primary)
+- L3 - RMF_AudioCapture_Open_Type(Auxiliary)
+- L3 - RMF_AudioCapture_GetStatus
+- L3 - RMF_AudioCapture_GetDefaultSettings
+- L3 - RMF_AudioCapture_GetCurrentSettings
+- L3 - RMF_AudioCapture_Start
+- L3 - RMF_AudioCapture_Stop
+- L3 - RMF_AudioCapture_Close
+
+## Raft L3 Testing Requirements
+
+- use these ut-core functions to achieve requirements
+
+### Internal CheckAudioDecodingCorrectFormat( Type Stream Type )
+
+- configuration YAML file contains all info on fixed streams and the format of each stream.
+- Set up the primary audio decoder, with a fixed stream.
```

Comment:
Pre-Conditions : 
Setup of a gstreamer pipeline to decode fixed audio streams from a given input source.
The audio pipeline must set up all the components required to activate the mixer.
RMF Audio taps off the mixer to capture it's data, this will be setup by the L3 test during the test setup.

Reply:
Done